modules/indexer/issues/meilisearch/meilisearch.go

```diff
@@ -223,6 +226,31 @@ func (b *Indexer) Search(ctx context.Context, options *internal.SearchOptions) (
 
 	hits := make([]internal.Match, 0, len(searchRes.Hits))
 	for _, hit := range searchRes.Hits {
+		if !options.IsFuzzyKeyword {
+			// as meilisearch does not have a non-fuzzy search and you can only change "typo tolerance" per index we have to post-filter the results
+			// https://www.meilisearch.com/docs/learn/configuration/typo_tolerance#configuring-typo-tolerance
+			// TODO: remove once https://github.com/orgs/meilisearch/discussions/377 is addressed
+			keyword := strings.ToLower(options.Keyword)
+			title, _ := hit.(map[string]any)["title"].(string)
+			if !strings.Contains(strings.ToLower(title), keyword) {
+				content, _ := hit.(map[string]any)["content"].(string)
+				if !strings.Contains(strings.ToLower(content), keyword) {
+					comments, _ := hit.(map[string]any)["comments"].([]any)
```

Comment:
Are these casts crash-safe?

Reply:
they are, should I let them fail if i don't get what i expect?

Reply:
What is the more commonly useful search, fuzzy or not? Fall back to that.

Reply:
right now we have just non-fuzzy for issues

-> https://github.com/go-gitea/gitea/pull/29671#discussion_r1518165048

I'll make it fail hard ...